modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -31,33 +38,54 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 	<aui:script>
 		var hasEventAttached = false;
+		var refreshCaptcha = document.getElementById('<portlet:namespace />refreshCaptcha');
+		var captcha = document.getElementById('<portlet:namespace />captcha');
+		var simpleCaptchaURL = '/o/captcha/v1.0/simple?portletId=<%= portletId %>';
```

Comment:
get from theme display

Reply:
I think that we can not get portletid from themedisplay. There are some ways to get portletId inside jsp file but they seem not to be simpler than the way we send from Java file.